tools/droplets/fabric/fabfile.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import fabric
+from fabric.api import run, put, puts, task, env
+import os
+
+env.user = "root"
+
+def clean_up() -> None:
+    """
+    Clean up remote machine before taking snapshot.
+    """
+    run("rm -rf /tmp/* /var/tmp/*")
+    run("history -c")
+    run("cat /dev/null > /root/.bash_history")
+    run("unset HISTFILE")
+    run("apt-get -y autoremove")
+    run("apt-get -y autoclean")
+    run(br"find /var/log -mtime -1 -type f -exec truncate -s 0 {} \;")
+    run("rm -rf /var/log/*.gz /var/log/*.[0-9] /var/log/*-????????")
+    run("rm -rf /var/lib/cloud/instances/*")
+    run("rm -rf /var/lib/cloud/instance")
+    puts("Removing keys...")
+    run("rm -f /root/.ssh/authorized_keys /etc/ssh/*key*")
+    run("dd if=/dev/zero of=/zerofile; sync; rm /zerofile; sync")
+    run("cat /dev/null > /var/log/lastlog; cat /dev/null > /var/log/wtmp")
+
+def install_files() -> None:
+    """
+    Install files onto remote machine.
+    Walk through the files in the "files" directory and copy them to the build system.
+    File permissions will be inherited.  If you need to change permissions on uploaded files
+    you can do so in a script placed in the "scripts" directory.
+    """
+    print("--------------------------------------------------")
+    print("Copying files in ./files to remote server")
+    print("--------------------------------------------------")
+    rootDir = './files'
+    for dirName, subdirList, fileList in os.walk(rootDir):
+        cDir = dirName.replace("./files", "")
+        print("Entering Directory: {}".format(cDir))
+        if cDir:
+            run("mkdir -p {}".format(cDir))
+        for fname in fileList:
+            cwd = os.getcwd()
+            rpath = cDir + "/" + fname
+            lpath = cwd + "/files" + cDir + "/" + fname
+            print('Moving File: {}'.format(lpath))
+            put(lpath, rpath, mirror_local_mode=True)
```

Comment:
Some line like this used to be `put(lpath,rpath,mirror_local_mode=True)` (without proper spaces)